RSSReader/Services/NetworkManager.m

```diff
@@ -0,0 +1,32 @@
+//
+//  NetworkManager.m
+//  RSSReader
+//
+//  Created by Владислав Станкевич on 19.11.20.
+//
+
+#import "NetworkManager.h"
+#import "FeedItem.h"
+
+@implementation NetworkManager
+
+- (void)loadFeedWithCompliteon:(void (^)(NSData*))completion {
+    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:@"http://news.tut.by/rss/index.rss"]
+                                                           cachePolicy:NSURLRequestUseProtocolCachePolicy
+                                                       timeoutInterval:10.0];
+    NSURLSession *session = [NSURLSession sharedSession];
+    NSURLSessionDataTask *dataTask = [session dataTaskWithRequest:request
+                                                completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+        if (error) {
+        } else {
+            //    NSError *e = nil;
+            dispatch_async(dispatch_get_main_queue(), ^{
+                completion(data);
+            });
```

Comment:
- switch to main queue inside "completion(data)"?

Reply:
In this case that would hidden and unexpected behavior. Completion should perform some task. The caller is to decide when to call it and which queue to perform the task on. I would suggest to keep it as it is.